src/event/event-type.js

```diff
@@ -180,6 +180,10 @@ const CustomEventType: EventTypes = {
    * Fires on the first time that playback has been started.
    */
   PLAYBACK_STARTED: 'playbackstarted',
+  /**
+   * Fires when the playback (includes postrolls) is ended.
+   */
+  PLAYBACK_ENDED: 'playbackended',
```

Comment:
Sounds as the Inverse of `PLAYBACK_STARTED` but it's not...

Reply:
`PLAYBACK_STARTED` is ok where it is.
It means that actual video frame is being rendered on screen.
Now that we have the ads playback taken into account it just means that we need to listen to ad_playing and fire it as well.
We already have a boolean flag to fire it only on first time so just extract this to a method, that is called from playing event and ad_playing event - first one reaching it will emit it and set the boolean flag to true.